xa/java/jms/department-helidon-teq/src/main/java/com/oracle/mtm/sample/data/AccountsService.java

```diff
@@ -0,0 +1,200 @@
+            throw new RuntimeException(e);
+        }
+        PreparedStatement statement = connection.prepareStatement(query);
+        statement.setDouble(1, amount);
+        statement.setString(2, accountId);
+        boolean res = statement.executeUpdate() > 0;
+
+        publishEvent(accountId, amount);
+
+        return res;
+    }
+
+    /**
+     * Deposit amount to an account
+     * @param accountId Account identity
+     * @param amount The amount to be deposited into the account
+     * @return boolean to indicate if the deposit was successful
+     * @throws SQLException
+     */
+    @Override
+    public boolean deposit(String accountId, double amount) throws SQLException, JMSException {
+        String query ="UPDATE accounts SET amount=amount+? where account_id=?";
+        Connection connection = null;
+        try {
+            connection = ((AQjmsSession) xaTopicSession).getDBConnection();
+        } catch (JMSException e) {
+            throw new RuntimeException(e);
+        }
+        PreparedStatement statement = connection.prepareStatement(query);
+        statement.setDouble(1, amount);
+        statement.setString(2, accountId);
+        boolean res = statement.executeUpdate() > 0;
+
+        publishEvent(accountId, amount);
+
+        return res;
+    }
+
+    /**
+     * Get balance amount from the account
+     * @param accountId Account identity
+     * @return Returns the balance associated with the account
+     * @throws SQLException
+     */
+    @Override
+    public double getBalance(String accountId) throws SQLException {
+        String query ="SELECT amount FROM accounts where account_id=?";
+        Connection connection = null;
+        try {
+            connection = ((AQjmsSession) xaTopicSession).getDBConnection();
+        } catch (JMSException e) {
+            throw new RuntimeException(e);
+        }
+        PreparedStatement statement = connection.prepareStatement(query);
+        statement.setString(1, accountId);
+        ResultSet dataSet = statement.executeQuery();
+        if (dataSet.next()) {
+            return Double.parseDouble(dataSet.getString("amount"));
+        }
+        statement.close();
+        throw new IllegalArgumentException("Account not found");
+    }
+
+    @Override
+    public void publishEvent(String accountId, double amount) throws JMSException, SQLException {
+        Topic topic = ((AQjmsSession) xaTopicSession).getTopic(user, topicName);
+        AQjmsTopicPublisher publisher = (AQjmsTopicPublisher) ((AQjmsSession) xaTopicSession).createPublisher(topic);
+        String msg = "Deposited: " + amount + " in account: " + accountId;
+        AQjmsTextMessage message = (AQjmsTextMessage) xaTopicSession.createTextMessage(msg);
+        publisher.publish(message, new AQjmsAgent[]{new AQjmsAgent("my_subscription", null)});
+        config.getLogger().log(Level.INFO, "Message published:" + msg);
+        // clean up
+        publisher.close();
```

Comment:
should we move this to finally ? or other option is we can check if AQjmsTopicPublisher is AutoClosable

Reply:
Make sense. Wrapped it inside try with auto closable block